### PR TITLE
[mono] Add Vector128 Negate and OnesComplement intrinsics for amd64

### PR DESCRIPTION
**Pull request by matouskozak**
Add support for the following Vector128 API's:

- Negate
- OnesComplement
- UnaryNegation operator (-)
- OnesComplement operator (~)

### CODE REVIEW

**Comment by matouskozak** [comment]
Duplication of code between amr64 and amd64 https://github.com/dotnet/runtime/issues/75144

**Review by vargaz — APPROVED** [review]

File: src/mono/mono/mini/mini-llvm.c [hunk 1]
```diff
@@ -11276,6 +11276,33 @@ MONO_RESTORE_WARNING
 		}
 #endif
 
+#ifdef TARGET_AMD64
+		case OP_AMD64_NEGATION: {
+			gboolean is_float = FALSE; 
+			if ( ins->inst_c1 == MONO_TYPE_R4 || ins->inst_c1 == MONO_TYPE_R8 )
```
Comment (vargaz) [inline src/mono/mono/mini/mini-llvm.c:11282]: This could be just:
gboolean is_float = (ins->inst_c1 == ...

File: src/mono/mono/mini/mini-llvm.c [hunk 2]
```diff
@@ -11276,6 +11276,33 @@ MONO_RESTORE_WARNING
...
 #endif
 
+#ifdef TARGET_AMD64
+		case OP_AMD64_NEGATION: {
+			gboolean is_float = FALSE; 
+			if ( ins->inst_c1 == MONO_TYPE_R4 || ins->inst_c1 == MONO_TYPE_R8 ) 
+				is_float = TRUE; 
+			
+			LLVMValueRef result; 
+			if ( is_float )
```
Comment (vargaz) [inline src/mono/mono/mini/mini-llvm.c:11286]: There is no space after ( in the mono coding convention.

File: src/mono/mono/mini/mini-llvm.c (hunk 1 above)
Comment (matouskozak) [reply src/mono/mono/mini/mini-llvm.c:11282]: Maybe I could move it completely to the `if` on line 11286 and not use `is_float` variable at all?

**Comment by fanyang-mono** [comment]
There is some missing test coverage for runtime tests running with LLVM AOT on x64. There is an open PR fixing this (#75645). Thus, adding `NO-MERGE` flag.

File: src/mono/mono/mini/mini-ops.h [hunk 3]
```diff
@@ -1778,5 +1773,8 @@ MINI_OP(OP_WASM_ONESCOMPLEMENT, "wasm_onescomplement", XREG, XREG, NONE)
 #endif
 
 #if defined(TARGET_ARM64) || defined(TARGET_AMD64)
-MINI_OP3(OP_BSL, "bitwise_select", XREG, XREG, XREG, XREG)
+MINI_OP3(OP_BSL,            "bitwise_select", XREG, XREG, XREG, XREG)
+MINI_OP(OP_NEGATION,        "negate", XREG, XREG, NONE)
+MINI_OP(OP_NEGATION_SCALAR, "negate_scalar", XREG, XREG, NONE)
+MINI_OP(OP_ONES_COMPLEMENT,  "ones complement", XREG, XREG, NONE)
```
Comment (fanyang-mono) [inline src/mono/mono/mini/mini-ops.h:1779]:
```suggestion
MINI_OP(OP_ONES_COMPLEMENT, "ones_complement", XREG, XREG, NONE)
```

**Review by fanyang-mono — COMMENTED** [review]
Mostly look good. Just one format change is needed before merge.

**Review by fanyang-mono — APPROVED** [review]

**Pull request merged**